### ISSUE TITLE
Update Interpreter

### ISSUE THREAD

**Issue by DavePearce** [Whiley/WhileyCompiler#477]
The interpreter was submitted by Sam from Waikato, but it needs to be updated.  In particular, to support the new nested block structure.

**Comment by samminweng** [comment]
Hi Dave,

I read the post about the new version of Whiley compiler. It seems a little unstable?

Should I upgrade the Whiley compiler to version 0.3.32 now or wait for the stable version?
What do you think?

**Comment by DavePearce** [comment]
Hi Sam,

> I read the post about the new version of Whiley compiler. It seems a
> little unstable?

Yeah, it's not ideal.  Most of the problems are related to the 
verification side of things.

> Should I upgrade the Whiley compiler to version 0.3.32 now or wait for
> the stable version?
> What do you think?

Tough call.  I would probably wait unless you have a compelling reason. 
  It will almost certainly mean a lot of changes to your own code as 
well, and there could be further changes down the track.  Personally, I 
would focus on your research as much as possible for now and wait until 
I finally release version 0.4.0.  That should be the best and most stable.

In releasing version 0.4.0 it looks likely that I am going to 
significantly cut the language down more.  For example, removing 
strings, sets, and maps.  This is to simplify things as much as 
possible.  Strings will then be implemented just as lists of integers. 
I may further reduce the number of operators and statements available as 
well.  For example, removing the "for" statement and the list append and 
sublist operators.

Basically, I am really struggling to deliver something that works well. 
  Therefore, my plan is cut out everything that is not essential for now 
at least, and deliver a smaller language that at least works well. 
Then, in the future, I'll put things I want back into it.

Cheers!

Dave

>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Whiley/WhileyCompiler/issues/477#issuecomment-76883263>.
>

-- 
Senior Lecturer in Computer Science,
School of Engineering and Computer Science
Victoria University of Wellington,
PO Box 600,
Wellington,
New Zealand.

Office: Cotton 231
Telephone: +64 (0)4 463 5833
URL: http://ecs.victoria.ac.nz/~djp

**Comment by samminweng** [comment]
Hi Dave, 

Thanks for the reply. I will wait for the reliable 0.4.0 version. 

Best regards, 
Sam
